### PR TITLE
Cart page product links.

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Added product links to the product listings on the cart page.

## Related Issue
Closes [PWA-740](https://jira.corp.magento.com/browse/PWA-740)

### Verification Stakeholders
@jimbo 
@soumya-ashok 
@schensley 

### Verification Steps
1. Go to the deployed app and add an item to the cart.
2. Go to the cart page and click on the item in the cart. It should take you to the product's PDP page.

## Checklist

- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2557.pwa-venia.com : LH Performance Expected 0.85 Actual 0.52, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 83.333333333333
https://pr-2557.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2557.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.41, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 50

**Comment by soumya-ashok** [comment]
@revanth0212 The image and product title links are working as intended. UX approved.

The checkout button on the cart page remains disabled for a second or two after the cart page loads before it displays as enabled. Any idea why?

**Comment by revanth0212** [comment]
> @revanth0212 The image and product title links are working as intended. UX approved.
> 
> The checkout button on the cart page remains disabled for a second or two after the cart page loads before it displays as enabled. Any idea why?

Yeah, I have noticed it as well. The button will be disabled until the GQL query is completed. This is the case in `develop` as of today.

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.js [hunk 1]
```diff
@@ -67,34 +77,44 @@ const Product = props => {
...
-                alt={name}
-                classes={{ image: classes.image, root: classes.imageContainer }}
-                width={IMAGE_SIZE}
-                resource={image}
-            />
-            <div className={classes.details}>
-                <span className={classes.name}>{name}</span>
-                <ProductOptions
-                    options={options}
+            <Link to={itemLink} className={classes.item}>
```
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/venia-ui/lib/components/CartPage/ProductListing/product.js:87]: Agreed, I would expect to see `text-decoration: underline` or some common treatment to indicate this is clickable. Looks nice doing keyboard nav, so lets just loop in UX and confirm what they want here. I usually expect names and images to be clickable, but not the entire content area.

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.css [hunk 2]
```diff
@@ -1,8 +1,16 @@
 .root {
     display: grid;
     grid-gap: 0.5rem 1rem;
-    grid-template-areas: 'image details kebab';
-    grid-template-columns: 100px 1fr min-content;
+    grid-template-areas: 'item quantity kebab';
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/CartPage/ProductListing/product.css:4]: This change broke the responsive flow of quantity below options, will need to be resolved.

Screen Shot 2020-07-20 at 8 48 15 AM

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/venia-ui/lib/components/CartPage/ProductListing/product.js:87]: Just reviewed #2549 and saw that this pattern seems to UX approved. I'll still mention this in standup, at the very least I think it needs a hover treatment so it looks as clickable with mouse as it does with keyboard nav.

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.css (hunk 2 above)
Comment (revanth0212) [reply packages/venia-ui/lib/components/CartPage/ProductListing/product.css]: My bad. Fixed it in bf8f2c3.

image

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.js [hunk 3]
```diff
@@ -84,37 +84,41 @@ const Product = props => {
...
                             options: classes.options,
                             optionLabel: classes.optionLabel
                         }}
                     />
-                    <span className={classes.price}>
-                        <Price currencyCode={currency} value={unitPrice} />
-                        {' ea.'}
-                    </span>
+                </Link>
+                <Link to={itemLink} className={classes.price}>
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CartPage/ProductListing/product.js:111]: Just to put it out there, I would only expect the image and the product name to be links to the PDP but not the options or price.

Comment (revanth0212) [reply packages/venia-ui/lib/components/CartPage/ProductListing/product.js:111]: I am awaiting UX's response on #2549 which is similar to this. Will update the comment here accordingly once I have an update.

**Comment by tjwiebell** [comment]
@revanth0212 - I would say having the entire area clickable or the two links @sirugh are both acceptable solutions, but individual links is a little odd. It's fine if you're just using this as an example to show UX, but this doesn't seem expected.

~[](https://gyazo.com/39a85e85f00ad5f313e3fee04f1e2056.gif)

**Comment by revanth0212** [comment]
> @revanth0212 - I would say having the entire area clickable or the two links @sirugh suggested are both acceptable solutions, but individual links is a little odd (even more so with configurable options). It's fine if you're just using this as an example to show UX, but this doesn't seem expected.
> 
> 

Totally agree with you @tjwiebell. That is actually the reason why I had a single Link previously but due to the UX requirements / Grid layout, I can not wrap the components in a single Link anymore. If I do so, the Quantity component will be wrapped in the Link as well which is not acceptable. My CSS might be rough but that is my understanding, I am open for suggestions.

**Comment by revanth0212** [comment]
Made changes to only have the Image and the name as links to the PDP page.

**Comment by soumya-ashok** [comment]
@revanth0212 I cleared my cache and checked the cart page - the product image and product title are linking correctly to the PDP and the rest of the text is static. UX approved.

**Comment by dpatil-magento** [comment]
@revanth0212 Please check this observation, I could not able to reproduce this on `develop`

Pre-condition - Test in Mobile device (I  checked in Android - chrome), works fine in desktop.
1. Add configurable product to cart.
2. Go to Cart page.
3. Go to Product Edit drawer.
4. Close the the edit drawer by tapping close icon or by hitting Update button without making any changes.

Issue -  Cart page ui is broken. Only product image is displayed.

image

**Comment by dpatil-magento** [comment]
> @revanth0212 Please check this observation, I could not able to reproduce this on `develop`
> 
> Pre-condition - Test in Mobile device (I checked in Android - chrome), works fine in desktop.
> 
> 1. Add configurable product to cart.
> 2. Go to Cart page.
> 3. Go to Product Edit drawer.
> 4. Close the the edit drawer by tapping close icon or by hitting Update button without making any changes.
> 
> Issue - Cart page ui is broken. Only product image is displayed.
> 
> image

Ignore this!! it's happening in `develop` too.  I will create separate bug.

**Comment by revanth0212** [comment]
> @revanth0212 Please check this observation, I could not able to reproduce this on `develop`
> 
> Pre-condition - Test in Mobile device (I checked in Android - chrome), works fine in desktop.
> 
> 1. Add configurable product to cart.
> 2. Go to Cart page.
> 3. Go to Product Edit drawer.
> 4. Close the the edit drawer by tapping close icon or by hitting Update button without making any changes.
> 
> Issue - Cart page ui is broken. Only product image is displayed.
> 

Wow hatsoff to you testing dude.. I have never seen that. Let's create a new bug ticket to handle this.

**Pull request merged**